modules/eks/argocd/main.tf

```diff
@@ -1,17 +1,18 @@
 locals {
-  enabled                = module.this.enabled
-  github_webhook_enabled = local.enabled && var.github_webhook_enabled
-  kubernetes_namespace   = var.kubernetes_namespace
-  oidc_enabled           = local.enabled && var.oidc_enabled
-  oidc_enabled_count     = local.oidc_enabled ? 1 : 0
-  saml_enabled           = local.enabled && var.saml_enabled
+  enabled = module.this.enabled
+
+  kubernetes_namespace = var.kubernetes_namespace
+  oidc_enabled         = local.enabled && var.oidc_enabled
+  oidc_enabled_count   = local.oidc_enabled ? 1 : 0
+  saml_enabled         = local.enabled && var.saml_enabled
   argocd_repositories = local.enabled ? {
-    for k, v in var.argocd_repositories : module.argocd_repo[k].outputs.repository => {
+    for k, v in var.argocd_repositories : replace(k, "/", "-") => {
```

Comment:
Can we leave a reference with `remote-state`?

Reply:
answered in slack thread. copy paste:


this is still using remote state with argocd-repo
```
  argocd_repositories = local.enabled ? {
    for k, v in var.argocd_repositories : replace(k, "/", "-") => {
      clone_url         = module.argocd_repo[k].outputs.repository_ssh_clone_url
      github_deploy_key = data.aws_ssm_parameter.github_deploy_key[k].value
      repository        = module.argocd_repo[k].outputs.repository
    }
  } : {}
```
the difference is that the key used for the local.argocd_repositories map is the name of the remote component (with a character replacement) instead of module.argocd_repo[k].outputs.repository. Now that value is given to the map as a value for repository instead